src/datasources/cache/redis.cache.service.ts

```diff
@@ -38,32 +40,39 @@ export class RedisCacheService
       return;
     }
 
+    const key = `${this.keyPrefix}${cacheDir.key}`;
+
     try {
-      await this.client.hSet(cacheDir.key, cacheDir.field, value);
-      await this.client.expire(cacheDir.key, expireTimeSeconds);
+      await this.client.hSet(key, cacheDir.field, value);
+      await this.client.expire(key, expireTimeSeconds);
     } catch (error) {
-      await this.client.hDel(cacheDir.key, cacheDir.field);
+      await this.client.hDel(key, cacheDir.field);
       throw error;
     }
   }
 
   async get(cacheDir: CacheDir): Promise<string | undefined> {
-    return await this.client.hGet(cacheDir.key, cacheDir.field);
+    const key = `${this.keyPrefix}${cacheDir.key}`;
+    return await this.client.hGet(key, cacheDir.field);
   }
 
   async deleteByKey(key: string): Promise<number> {
+    const keyWithPrefix = `${this.keyPrefix}${key}`;
     // see https://redis.io/commands/unlink/
-    const result = await this.client.unlink(key);
+    const result = await this.client.unlink(keyWithPrefix);
     await this.set(
-      new CacheDir(`invalidationTimeMs:${key}`, ''),
+      new CacheDir(`invalidationTimeMs:${keyWithPrefix}`, ''),
```

Comment:
There's a bug in this implementation where the prefix will be added again because we are calling `set` here which also adds the prefix.

Reply:
Fixed in [20dc54f](https://github.com/safe-global/safe-client-gateway/pull/998/commits/20dc54f91051f3a9afceaaffdaa9cdf19f91a11a)